test/math.cpp

```diff
@@ -30,57 +34,47 @@ typedef std::complex<double> complex_double;
 
 template<typename T>
 T sigmoid(T in) {
-    return 1.0 / (1.0 + std::exp(-in));
+    return T(1.0 / (1.0 + std::exp(-in)));
 }
 
 template<typename T>
 T rsqrt(T in) {
-    return 1.0/sqrt(in);
+    return T(1.0 / sqrt(in));
 }
 
-#define TEST_REAL(T, func, err, lo, hi)                          \
-    TEST(MathTests, Test_##func##_##T) {                         \
-        try {                                                    \
-            SUPPORTED_TYPE_CHECK(T);                             \
-            af_dtype ty = (af_dtype)dtype_traits<T>::af_type;    \
-            array a     = (hi - lo) * randu(num, ty) + lo + err; \
-            eval(a);                                             \
-            array b = func(a);                                   \
-            vector<T> h_a(a.elements());                         \
-            vector<T> h_b(b.elements());                         \
-            a.host(&h_a[0]);                                     \
-            b.host(&h_b[0]);                                     \
-                                                                 \
-            for (int i = 0; i < num; i++) {                      \
-                ASSERT_NEAR(h_b[i], func(h_a[i]), err)           \
-                    << "for value: " << h_a[i] << endl;          \
-            }                                                    \
-        } catch (exception & ex) { FAIL() << ex.what(); }        \
+#define TEST_REAL(T, func, err, lo, hi)                                     \
+    TEST(MathTests, Test_##func##_##T) {                                    \
+        try {                                                               \
+            SUPPORTED_TYPE_CHECK(T);                                        \
+            af_dtype ty = (af_dtype)dtype_traits<T>::af_type;               \
+            array a     = (hi - lo) * randu(num, ty) + lo + err;            \
+            a           = a.as(ty);                                         \
```

Comment:
Is this explicit necessary ? The reason I ask that is if this line isn't there, then `TEAT_REAL` and `TEST_CPLX` are identical.

Reply:
No, this is not necessary. It was required earlier because we were using the new ASSERT_VEC_ARRAY_NEAR macro.